### PR TITLE
Add vectorized RandomBrightness, RandomContrast, RandomHue and RandomColorJitter

### LINKED ISSUE #1376: Vectorize RandomContrast
Based on PR https://github.com/keras-team/keras-cv/pull/1373 we would like to vectorize RandomContrast.

One issue with this is that `tf.image.adjust_contrast()` does not accept a batch argument for contrast adjustment, so we'll have to roll a pure TF version.

### LINKED ISSUE #1405: Vectorize RandomBrightness, RandomContrast, RandomHue and RandomColorJitter
**Short Description**

As #1392 was merged, I'm willing to contribute for vectorized `RandomBrightness`, `RandomContrast`, `RandomHue` and finally to complete vectorized `RandomColorJitter`.

`RandomColorJitter` is widely adopted, so the improvement from vectorization should benefit the community.

**Other Information**

There is an unsolved conversation at https://github.com/keras-team/keras-cv/pull/1392#discussion_r1107966121
and my questions are:
1. Should we move A/B testings from unit tests into `./benchmarks/random_*.py`?
2. `./benchmarks/` might be flooded by these scripts. Should we create a new folder for these?

For 1., take `./benchmarks/vectorized_random_saturation.py` as example:
```python
# add old implementation
class OldRandomSaturation(BaseImageAugmentationLayer):
    ...

# add tests
class RandomSaturationTest(tf.test.TestCase):
    def test_correctness_with_tf_adjust_saturation_normalized_range(self):
        ...
        self.assertAllClose(old_output, output, atol=1e-5, rtol=1e-5)

    def test_correctness_with_tf_adjust_saturation_rgb_range(self):
        ...
        self.assertAllClose(old_output, output, atol=1e-3, rtol=1e-5)


if __name__ == "__main__":
    # run benchmarks (get some figures for improvement)
    ...
    # run unit tests
    tf.test.main()
```

For 2., I think something like `./benchmarks/vectorization_strategy/vectorized_*.py` can be a better layout?

### PR DESCRIPTION
**Pull request by james77777778**
# What does this PR do?

Fixes #1405 
Fixes #1376 

Following #1392 but I moved A/B testing into benchmarks suggested by @ianstenbit 's comment at https://github.com/keras-team/keras-cv/pull/1392#discussion_r1107523531, also I added XLA Mode `jit_compile=True` for benchmarks suggested by @bhack 

For unit tests, only `keras_cv/layers/preprocessing/random_hue_test.py` failed at `test_adjust_full_opposite_hue` and could be resolved by relaxing constraints `atol=1e-5, rtol=1e-5`

For numerical check (used `image_shape = (1024, 32, 32, 3)` locally):
- `./benchmarks/vectorized_random_brightness.py` passes without any constraint relaxing
- `./benchmarks/vectorized_random_contrast.py` passes with `atol=1e-5, rtol=1e-5` at rgb value range
- `./benchmarks/vectorized_random_hue.py` passes with `atol=1e-3, rtol=1e-5` at rgb value range

`./benchmarks/vectorized_random_color_jitter.py` has no numerical check because it is composited by above layers

Some notes:
- `OldRandomBrightness` got similar performance with Graph/XLA Mode, other vectorized layers got significant boost
- `RandomHue` failed to run XLA mode
- The benchmark of `RandomColorJitter` was done after I vectorizing all sub-layers

This is results from benchmarks:

#### RandomBrightness
<details>

comparison
comparison_no_old_eager
</details>

#### RandomContrast
<details>

comparison
comparison_no_old_eager
</details>

#### RandomHue (no XLA Mode)
<details>

comparison
comparison_no_old_eager
</details>

#### RandomColorJitter
<details>

comparison
comparison_no_old_eager
</details>

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [X] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?
@LukeWood @ianstenbit

### CODE REVIEW

**Review by ianstenbit — APPROVED** [review]
@james77777778 this is great -- thank you!

A few quick questions:

For RandomHue -- did XLA also fail for the original implementation? Or only for the new vectorized implementation?

For RandomBrightness -- the graph seems to be missing a few of the trials. I assume this is because the scale of the Y axis is too small? Would you mind updating that?

Thank you for this great contribution!

**Review by LukeWood — APPROVED** [review]

**Comment by james77777778** [comment]
> For RandomHue -- did XLA also fail for the original implementation? Or only for the new vectorized implementation?

- RandomHue (vectorized): ran w/o problem
- OldRandomHue: failed for XLA

Code snippet to bypass failure
```python
        # XLA Mode
        if aug is OldRandomHue:
            print("Skip XLA Mode")
            continue
        c = aug.__name__ + " XLA Mode"
        layer = aug(**aug_args)
```

comparison_

> For RandomBrightness -- the graph seems to be missing a few of the trials. I assume this is because the scale of the Y axis is too small? Would you mind updating that?
> 

Yes, the Y axis is too small to distinguish the differences.

From logs:
|Mode|N|RandomBrightness|OldRandomBrightness|
|-|-|-|-|
|Eager|1000|0.0117|0.0818|
||2000|0.0135|0.0866|
||5000|0.0611|0.1357|
||10000|0.1203|0.1954|
|Graph|1000|0.0105|0.0072|
||2000|0.0116|0.0134|
||5000|0.0604|0.0604|
||10000|0.1199|0.1201|
|XLA|1000|0.0072|0.0073|
||2000|0.0117|0.0137|
||5000|0.0608|0.0609|
||10000|0.1201|0.1213|

The vectorized RandomBrightness won in Eager mode but got similar performance in Graph/XLA mode.

Thanks for the review!
